bin/cheribsdtest/cheribsdtest.h

```diff
@@ -311,10 +311,11 @@ _cheribsdtest_check_errno(const char *context, int actual, int expected)
 #define CHERIBSDTEST_CHECK_CALL_ERROR(call, expected_errno)		\
 	do {								\
 		errno = 0;						\
-		int __ret = call;					\
+		__typeof(call) __ret = call;				\
 		int call_errno = errno;					\
-		CHERIBSDTEST_VERIFY2(__ret == -1,			\
-		    #call " unexpectedly returned %d", __ret);		\
+		CHERIBSDTEST_VERIFY2(__ret == (__typeof(__ret))-1,	\
```

Comment:
I think I'd rather just have a separate macro that use %#p to use with calls to mmap().  Maybe you can use C11 _Generic() to make this more automatic though?  One of the results of this change is that integer return values that are signed (`int`) are now always printed as unsigned values.

Reply:
I've restored %d as the default using _Generic. It's a little duplicative due to interactions between _Generic and the preprocessor, but seems to work ok.